### DISCUSSION TITLE
Hello! When I drag the earth to a certain Angle, the picture will pass through the earth, what can I do to avoid this? Thank you!

### DISCUSSION THREAD

**Discussion by luohuiiqng** [pelicanmapping/osgearth#2432] [Support Q&A] [answered]
![6cee9f1557646aeb876d0fb12973bc7](https://github.com/gwaldron/osgearth/assets/82392471/f9f507f6-bae7-41c0-aeb9-f2a4c5c4c661)

**Comment by gwaldron** [comment, marked as answer]
This looks like clipping caused by the OpenGL near clipping plane. You can either try reducing the distance of the near plane in your projection matrix, or you can try using the `LogarithmicDepthBuffer` utility:
```c++
osgEarth::Util::LogarithmicDepthBuffer logDepth;
logDepth.install(viewer.getCamera());
```

**Reply by luohuiiqng** [reply]
Thanks, it works!
